feign-reactor-spring-configuration/src/main/java/reactivefeign/spring/config/ReactiveFeignBasicConfigurator.java

```diff
@@ -160,4 +189,24 @@ static ReactiveRetryPolicy configureRetryPolicyFromProperties(
 		}
 		return retryPolicy;
 	}
+
+	private static ReactiveHttpRequest reactiveHttpRequestWithQuery(ReactiveHttpRequest reactiveHttpRequest, String key, String value) {
```

Comment:
I this it's better to move out all code related to interceptors to ReactiveHttpRequestInterceptors

Create 
```
addQueries(){

for (Map.Entry<String, List<String>> entry : config.getDefaultQueryParameters().entrySet()) {
				// Every Map entry is gonna belong to it's own interceptor
				resultBuilder = resultBuilder.addRequestInterceptor(reactiveHttpRequest -> {
					for (String value : entry.getValue()) {
						reactiveHttpRequest = reactiveHttpRequestWithQuery(reactiveHttpRequest, entry.getKey(), value);
					}
					return Mono.just(reactiveHttpRequest);
				});
			}

}
```
and 
```
addHeaders(){
for (Map.Entry<String, List<String>> entry : config.getDefaultRequestHeaders().entrySet()) {
				// Every Map entry is gonna belong to it's own interceptor
				resultBuilder = resultBuilder.addRequestInterceptor(reactiveHttpRequest -> {
					reactiveHttpRequest.headers().put(entry.getKey(), entry.getValue());
					return Mono.just(reactiveHttpRequest);
				});
			}
}
```

Reply:
Extracted to `ReactiveHttpRequestInterceptors`. Now it looks really good, thanks!
More over, now I know about this useful class, can use it somewhere else in my project!